lib/component_factory.rb

```diff
@@ -78,16 +80,19 @@ def _transform_row(component, inner)
   def _transform_columns(component, inner)
 
     col_count = component.parent.elements.size
-    small_size = component.attributes['small'] || self.column_count
-    large_size = component.attributes['large'] || component.attributes['small'] || (self.column_count / col_count).to_i
+    small_attr = component.attribute('small')
+    large_attr = component.attribute('large')
+
+    small_size = small_attr ? small_attr.value : self.column_count
+    large_size = large_attr ? large_attr.value : small_size || (self.column_count / col_count).to_i
```

Comment:
I think this is actually changing behavior in a place that may not be intended... in particular if neither small nor large is set, we want the defaults to be different for small and large, and this makes them the same.  This seems like there is a missing test, perhaps you can add it? The test would look something like
```
<row>
  <columns>a</columns>
  <columns>b</columns>
</row>
```
and the result I believe should be something that has large size 6 and small size 12